src/main/angular/app/graph/graph.component.ts

```diff
@@ -33,6 +33,11 @@ import {DateRange} from '../../model/date_range';
   styleUrls: ['./graph.component.css'],
 })
 export class GraphComponent implements OnInit {
+  /** Informs the user to select a project. Displayed when none are shown. */
+  private static SELECT_PROJECT_MESSAGE =
+    'Please select a project below to generate a graph.';
+  private static LOADING_MESSAGE = 'Waiting on server, please wait.';
+
```

Comment:
could you make these constants, so that we know that they can't change?

Reply:
I can't have class members be constants, so I moved them to `constants.ts`